src/python/WMCore/WMSpec/StdSpecs/DataProcessing.py

```diff
@@ -46,9 +46,9 @@ def getWorkloadCreateArgs():
                     "PrimaryDataset": {"optional": True, "validate": primdataset,
                                        "attr": "inputPrimaryDataset", "null": True},
                     "RunBlacklist": {"default": [], "type": makeList, "null": False,
-                                     "validate": lambda x: all([int(y) > 0 for y in x])},
+                                     "validate": lambda x: all([isinstance(y, int) for y in x])},
```

Comment:
Valentin, I think we should actually keep the old validation as well. So the code that I would suggest to have in place is `isinstance(y, int) and int(y) > 0`, thus catching either the data type and a positive number.